dark/gor4.py

```diff
@@ -17,32 +17,25 @@
     @param secondaryFile: The C{str} name of a file containing known
         secondary structure information for the sequences in C{sequenceFile}.
     """
-    cdef cgor4.State *_state
 
-    def __cinit__(self, sequenceFile=_SEQUENCES, secondaryFile=_SECONDARY):
-        cdef int error
+    def __init__(self, sequenceFile=_SEQUENCES, secondaryFile=_SECONDARY):
         sequenceFile = sequenceFile.encode('UTF-8')
-        cdef char* c_sequenceFile = sequenceFile
         secondaryFile = secondaryFile.encode('UTF-8')
-        cdef char* c_secondaryFile = secondaryFile
 
-        self._state = cgor4.initialize(c_sequenceFile, c_secondaryFile,
-                                       &error)
-        if error:
-            raise Exception('Error in gor4-base.c initialize function.')
+        error = ffi.new('int *')
+        state = lib.initialize(ffi.new('char[]', sequenceFile),
+                               ffi.new('char[]', secondaryFile), error)
+        if state == 0:
+            raise Exception('Error in gor4 initialization.')
```

Comment:
Well, this was actually a bug. It should have been checking the `error` value, not `state`.

There doesn't need to be anything more specific here, as the GOR4 C code will print an error message to stderr if something goes wrong.  I just tried some non-existent filenames for the initialization to make sure that happens.